examples/cuda/riscv.mk

```diff
@@ -226,39 +233,55 @@ RISCV_INCLUDES += -I$(BSG_MANYCORE_DIR)/software/bsg_manycore_lib
 # Check for machine definitions. These are typically set by
 # Makefile.machine.include
 
-ifndef BSG_MACHINE_GLOBAL_X
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_GLOBAL_X undefined"))
-endif
-ifndef BSG_MACHINE_GLOBAL_Y
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_GLOBAL_Y undefined"))
-endif
-RISCV_DEFINES += -Dbsg_global_X=$(BSG_MACHINE_GLOBAL_X)
-RISCV_DEFINES += -Dbsg_global_Y=$(BSG_MACHINE_GLOBAL_Y)
-
-ifndef BSG_MACHINE_POD_TILES
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_POD_TILES undefined"))
-endif
-ifndef BSG_MACHINE_PODS_X
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_PODS_X undefined"))
-endif
-ifndef BSG_MACHINE_PODS_Y
-$(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_PODS_Y undefined"))
-endif
-RISCV_DEFINES += -Dbsg_group_size=$(BSG_MACHINE_POD_TILES)
-RISCV_DEFINES += -Dbsg_pods_X=$(BSG_MACHINE_PODS_X)
-RISCV_DEFINES += -Dbsg_pods_Y=$(BSG_MACHINE_PODS_Y)
+# ifndef BSG_MACHINE_GLOBAL_X
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_GLOBAL_X undefined"))
+# endif
+# ifndef BSG_MACHINE_GLOBAL_Y
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_GLOBAL_Y undefined"))
+# endif
+# RISCV_DEFINES += -Dbsg_global_X=$(BSG_MACHINE_GLOBAL_X)
+# RISCV_DEFINES += -Dbsg_global_Y=$(BSG_MACHINE_GLOBAL_Y)
+
+# ifndef BSG_MACHINE_POD_TILES
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_POD_TILES undefined"))
+# endif
+# ifndef BSG_MACHINE_PODS_X
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_PODS_X undefined"))
+# endif
+# ifndef BSG_MACHINE_PODS_Y
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_PODS_Y undefined"))
+# endif
+# RISCV_DEFINES += -Dbsg_group_size=$(BSG_MACHINE_POD_TILES)
+# RISCV_DEFINES += -Dbsg_pods_X=$(BSG_MACHINE_PODS_X)
+# RISCV_DEFINES += -Dbsg_pods_Y=$(BSG_MACHINE_PODS_Y)
+
+# ifndef BSG_MACHINE_HOST_X_CORD
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_HOST_X_CORD undefined"))
+# endif
+# ifndef BSG_MACHINE_HOST_Y_CORD
+# $(error $(shell echo -e "$(RED)BSG MAKE ERROR: BSG_MACHINE_HOST_Y_CORD undefined"))
+# endif
+# RISCV_DEFINES += -DIO_X_INDEX=$(BSG_MACHINE_HOST_X_CORD)
+# RISCV_DEFINES += -DIO_Y_INDEX=$(BSG_MACHINE_HOST_Y_CORD)
+# RISCV_DEFINES += -DPREALLOCATE=0
+# RISCV_DEFINES += -DHOST_DEBUG=0
+
+RISCV_DEFINES +=-Dbsg_tiles_X=$(bsg_tiles_X) -Dbsg_tiles_Y=$(bsg_tiles_Y)
```

Comment:
i think these can be derived from the Makefile.machine.include file.

Reply:
I don't really know ;)
I just copied from makefile.builddef